lib/new_relic/telemetry/redix.ex

```diff
@@ -0,0 +1,170 @@
+      connections: %{},
+      collect_db_query?: NewRelic.Config.feature?(:db_query_collection)
+    }
+
+    :telemetry.attach_many(
+      config.handler_id,
+      @init_events,
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    Process.flag(:trap_exit, true)
+    {:ok, config}
+  end
+
+  def handle_call({@redix_connection, meta}, _from, config) do
+    [host, port] = meta.address |> String.split(":")
+
+    config =
+      update_in(
+        config.connections,
+        &Map.put(
+          &1,
+          meta.connection,
+          %{address: meta.address, host: host, port: port}
+        )
+      )
+
+    :telemetry.detach(config.handler_id)
+
+    :telemetry.attach_many(
+      config.handler_id,
+      @connected_events,
+      &__MODULE__.handle_event/4,
+      config
+    )
+
+    {:reply, :ok, config}
+  end
+
+  def terminate(_reason, %{handler_id: handler_id}) do
+    :telemetry.detach(handler_id)
+  end
+
+  def handle_event(@redix_connection, _, meta, _config) do
+    GenServer.call(__MODULE__, {@redix_connection, meta})
+  end
+
+  def handle_event(
+        @redix_pipeline_stop,
+        %{duration: duration},
+        %{commands: commands, connection: connection} = meta,
+        config
+      ) do
+    end_time_ms = System.system_time(:millisecond)
+    duration_ms = System.convert_time_unit(duration, :native, :millisecond)
+    duration_s = duration_ms / 1000
+    start_time_ms = end_time_ms - duration_ms
+
+    datastore = "Redis"
+    {operation, query} = parse_command(commands, collect: config.collect_db_query?)
+
+    pid = inspect(self())
+    id = {:redix, make_ref()}
+    parent_id = Process.get(:nr_current_span) || :root
+
+    instance = config.connections[connection] || %{}
+
+    address = instance[:address] || "unknown"
+    hostname = instance[:host] || "unknown"
+    port = instance[:port] || "unknown"
+
+    metric_name = "Datastore/operation/#{datastore}/#{operation}"
+    secondary_name = "#{inspect(connection)} #{address}"
+
+    NewRelic.Transaction.Reporter.add_trace_segment(%{
+      primary_name: metric_name,
+      secondary_name: secondary_name,
+      attributes: %{
+        sql: query,
+        operation: operation,
+        host: hostname,
+        port_path_or_id: port
+      },
+      pid: pid,
+      id: id,
+      parent_id: parent_id,
+      start_time: start_time_ms,
+      end_time: end_time_ms
+    })
+
+    NewRelic.report_span(
+      timestamp_ms: start_time_ms,
+      duration_s: duration_s,
+      name: metric_name,
+      edge: [span: id, parent: parent_id],
+      category: "datastore",
+      attributes:
+        %{
+          component: datastore,
+          "span.kind": :client,
+          "db.statement": query,
+          "peer.address": address,
+          "peer.hostname": hostname,
+          "db.operation": operation,
+          "redix.connection": inspect(connection)
+        }
+        |> maybe_add("redix.error", meta[:error])
```

Comment:
* create a test situation that exercises this, I think it's changed

https://github.com/whatyouhide/redix/commit/3cd9beaa9a168fa3412d0d4b505b55ca76acc8cb#diff-d0c8798ead6df7f075d966a50c6d0fbbR110

Reply:
https://github.com/whatyouhide/redix/commit/3cd9beaa9a168fa3412d0d4b505b55ca76acc8cb#diff-f27faf01f440bea783cd29f2ccf82345R640-R641